packages/core/src/FormField.js

```diff
@@ -9,8 +9,10 @@ const FormField = props => {
   const children = React.Children.toArray(props.children)
   const [field] = children.filter(child => child.type.isField)
   const [label] = children.filter(child => child.type.isLabel)
+  const valueNoLabel = !label && field && !!field.props.value
   const showLabel =
-    (label && !label.props.autoHide) || (field && !!field.props.value)
+    ((label && !label.props.autoHide) || (field && !!field.props.value)) &&
+    !valueNoLabel
```

Comment:
Side thought: when these imperative object key conditions grows, it seems time to explore utility function or library to make code declarative, (e.g. Ramda). Maybe worth to log an issue.